shared/src/simulation/behaviors/assign-leader.ts

```diff
@@ -35,51 +42,189 @@ export const assignLeaderBehavior: SimulationBehavior<AssignLeaderBehaviorState>
+                    break;
+                case 'collectInformationEvent':
+                    // This behavior answerers queries for material, personnel and vehicles because the leader typically holds those information
+                    {
+                        // If there is no leader queries cant be answered
+                        if (!behaviorState.leaderId) {
+                            return;
+                        }
+                        switch (event.informationType) {
+                            case 'materialCount':
+                                {
+                                    const radiogram = getActivityById(
+                                        draftState,
+                                        simulatedRegion.id,
+                                        event.generateReportActivityId,
+                                        'generateReportActivity'
+                                    )
+                                        .radiogram as Mutable<MaterialCountRadiogram>;
+                                    const materials = Object.values(
+                                        draftState.materials
+                                    ).filter((material) =>
+                                        isInSpecificSimulatedRegion(
+                                            material,
+                                            simulatedRegion.id
+                                        )
+                                    );
+                                    const canCaterForRed = materials
+                                        .map(
+                                            (material) =>
+                                                material.canCaterFor.red
+                                        )
+                                        .reduce((a, b) => a + b, 0);
+                                    const canCaterForYellow = materials
+                                        .map(
+                                            (material) =>
+                                                material.canCaterFor.yellow
+                                        )
+                                        .reduce((a, b) => a + b, 0);
+                                    const canCaterForGreen = materials
+                                        .map(
+                                            (material) =>
+                                                material.canCaterFor.green
+                                        )
+                                        .reduce((a, b) => a + b, 0);
+                                    radiogram.materialForPatients.red =
+                                        canCaterForRed;
+                                    radiogram.materialForPatients.yellow =
+                                        canCaterForYellow;
+                                    radiogram.materialForPatients.green =
+                                        canCaterForGreen;
```

Comment:
In my opinion, you can remove the variables from above, as they serve no purpose other than to transfer one value once from a to b

Reply:
But whatever, I think we just think about code differently

Reply:
Done